home/views/data_sources.py

```diff
@@ -234,6 +235,9 @@ def _departments_data():
 
 class GenedData(View):
     def get(self, request):
+        if "geneds" not in request.GET:
+            return render(request, "404.html")
```

Comment:
you should be able to `raise Http404()` here instead of rendering the template directly. Let django handling the 404 error

Reply:
Whoops, forgot that was a thing! Done.